File: src.csharp/AlphaTab/Core/EcmaScript/Math.cs
using System;

namespace AlphaTab.Core.EcmaScript
{
    public static class Math
    {
        private static readonly Random Rnd = new Random();
        public static double PI => System.Math.PI;

        public static double Random()
        {
            return Rnd.NextDouble();
        }

        public static double Abs(double v)
        {
            return System.Math.Abs(v);
        }

        public static double Max(double a, double b)
        {
            return System.Math.Max(a, b);
        }

        public static double Min(double a, double b)
        {
            return System.Math.Min(a, b);
        }

        public static double Exp(double e)
        {
            return System.Math.Exp(e);
        }

        public static double Log(double d)
        {
            return System.Math.Log(d);
        }

        public static double Pow(double x, double y)
        {
            return System.Math.Pow(x, y);
        }

        public static double Floor(double d)
        {
            return System.Math.Floor(d);
        }

        public static double Round(double d)
        {
            return System.Math.Round(d);
        }

        public static double Sin(double a)
        {
            return System.Math.Sin(a);
        }

        public static double Log2(double d)
        {
            return System.Math.Log(d, 2);
        }

        public static double Sqrt(double d)
        {
            return System.Math.Sqrt(d);
        }

        public static double Ceil(double d)
        {
            return System.Math.Ceiling(d);
        }

        public static double Asin(double d)
        {
            return System.Math.Asin(d);
        }

        public static double Tan(double d)
        {
            return System.Math.Tan(d);
        }

        public static double Log10(double d)
        {
            return System.Math.Log10(d);
        }
    }
}

File: src.kotlin/alphaTab/src/commonMain/kotlin/alphaTab/core/ecmaScript/Math.kt
package alphaTab.core.ecmaScript

import kotlin.math.pow

class Math {
    companion object {
        public const val PI: Double = kotlin.math.PI
        public inline fun log10(x: Double): Double {
            return kotlin.math.log10(x)
        }

        public inline fun round(x: Double): Double {
            return kotlin.math.round(x)
        }

        public inline fun exp(x: Double): Double {
            return kotlin.math.exp(x)
        }

        public inline fun tan(x: Double): Double {
            return kotlin.math.tan(x)
        }

        public inline fun ceil(x: Double): Double {
            return kotlin.math.ceil(x)
        }

        public inline fun asin(x: Double): Double {
            return kotlin.math.asin(x)
        }

        public inline fun sqrt(x: Double): Double {
            return kotlin.math.sqrt(x)
        }

        public inline fun log(x: Double): Double {
            return kotlin.math.ln(x)
        }

        public inline fun sin(x: Double): Double {
            return kotlin.math.sin(x)
        }

        public inline fun pow(b: Double, e: Double): Double {
            return b.pow(e)
        }

        public inline fun abs(x: Double): Double {
            return kotlin.math.abs(x)
        }

        public inline fun floor(x: Double): Double {
            return kotlin.math.floor(x)
        }

        public inline fun log2(x: Double): Double {
            return kotlin.math.log2(x)
        }

        public inline fun min(a: Double, b: Double): Double {
            return kotlin.math.min(a, b)
        }

        public inline fun max(a: Double, b: Double): Double {
            return kotlin.math.max(a, b)
        }

        public inline fun random(): Double {
            return kotlin.random.Random.nextDouble()
        }
    }
}
